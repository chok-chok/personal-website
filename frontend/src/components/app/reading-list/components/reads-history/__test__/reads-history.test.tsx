import * as React from 'react';
import ReadsHistory, { ReadsHistoryProps } from '../reads-history';
import { shallow } from 'enzyme';

const props: ReadsHistoryProps = {
  books: [
    {
      title: 'Domain-Driven Design: Tackling Complexity in the Heart of Software',
      author: 'Eric Evans',
      coverImage:
        '//images.ctfassets.net/9iq7dqw56i6q/36gMyk68F3OfW1K9iBDvZV/defc330e0ae1fd27933e82117c13b3bd/domain_driven_design.jpg',
      rating: 4,
      readDate: new Date('December 17, 1995 03:24:00'),
      review: 'My review',
    },
    {
      title: 'Domain-Driven Design: Tackling Complexity in the Heart of Software',
      author: 'Eric Evans',
      coverImage:
        '//images.ctfassets.net/9iq7dqw56i6q/36gMyk68F3OfW1K9iBDvZV/defc330e0ae1fd27933e82117c13b3bd/domain_driven_design.jpg',
      rating: 4,
      readDate: new Date('December 17, 1995 03:24:00'),
    },
  ],
};

describe('Test ReadsHistory component', () => {
  it('should renders correctly', () => {
    const tree = shallow(<ReadsHistory books={props.books} />);

    expect(tree).toMatchSnapshot();
  });

  it('should only renders 2 books', () => {
    const wrapper = shallow(<ReadsHistory books={props.books} />);

    expect(wrapper.find('.finished-book')).toHaveLength(2);
  });
});
