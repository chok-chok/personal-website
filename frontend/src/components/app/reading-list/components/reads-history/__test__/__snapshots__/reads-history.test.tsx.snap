// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ReadsHistory component should renders correctly 1`] = `
<Fragment>
  <styled__StyledH2>
    What I have read
  </styled__StyledH2>
  <styled__FinishedBookWrapper>
    <FinishedBook />
    <FinishedBook />
    <FinishedBook />
  </styled__FinishedBookWrapper>
</Fragment>
`;
